src/plugins/intel_gna/legacy/include/legacy/graph_tools.hpp

```diff
@@ -192,7 +192,7 @@ inline void UnorderedDFS(std::unordered_set<CNNLayer*>& visited,
 
         // visit children
         for (auto& od : cnnLayer->outData) {
-            for (auto nl : getInputTo(od)) {
+            for (auto& nl : getInputTo(od)) {
```

Comment:
Isn't it better to use const auto& in all occurences as we are only reading the values, not modifying them?

Reply:
Good catch